edgelet/hsm-sys/azure-iot-hsm-c/src/edge_hsm_client_store.c

```diff
@@ -1692,40 +1705,171 @@ static int generate_edge_hsm_certificates_if_needed(void)
 static int get_tg_env_vars(char **trusted_certs_path, char **device_ca_path, char **device_pk_path)
 {
     int result;
+    char *tb_path = NULL, *cert_path = NULL, *pk_path = NULL;
 
-    if (hsm_get_env(ENV_TRUSTED_CA_CERTS_PATH, trusted_certs_path) != 0)
+    if (hsm_get_env(ENV_TRUSTED_CA_CERTS_PATH, &tb_path) != 0)
     {
         LOG_ERROR("Failed to read env variable %s", ENV_TRUSTED_CA_CERTS_PATH);
+        FREEIF(tb_path);
```

Comment:
Just `FREEIF` all three of them at the end if `result != 0`, instead of these six calls.